client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -341,9 +343,14 @@ private void throwIfClosed() throws TxnFailedException {
                 RuntimeException::new);
         UUID txnId = txnSegments.getTxnId();
         Map<Segment, SegmentTransaction<Type>> transactions = new HashMap<>();
+        DelegationTokenProvider tokenProvider = null;
```

Comment:
we can initialize it here removing the if block below . (same below too...)

Reply:
The reason I initialized it like this is to avoid having to initializing it on a per-segment basis. So, here, we initialize it once depending on whether the `tokenProvider` is null. 

https://github.com/pravega/pravega/blob/5d21e6a484657f8c9dc0a6c74d7337e8560bb19b/client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java#L346-L352